eb_hooks.py

```diff
@@ -965,52 +965,56 @@ def post_postproc_cuda(self, *args, **kwargs):
     Remove files from CUDA installation that we are not allowed to ship,
     and replace them with a symlink to a corresponding installation under host_injections.
     """
+    if self.name == 'CUDA':
```

Comment:
The change here is actually quite minimal. The previous condition was
```
if self.name == 'CUDA' and eessi_installation:
```
which meant that when someone tried to install CUDA but _wasn't_ doing an EESSI installation you would (incorrectly) get the `EasyBuildError`

Reply:
This change works as expected (tested using `EESSI-extend`):
```
ocaisa@~/EESSI/software-layer(hooks_check)$ module list

Currently Loaded Modules:
  1) EESSI/2023.06   2) EasyBuild/5.0.0   3) EESSI-extend/2023.06-easybuild

ocaisa@~/EESSI/software-layer(hooks_check)$ eb CUDA-Samples-12.1-GCC-12.3.0-CUDA-12.1.1.eb --dry-run --accept-eula-for=CUDA --hooks=eb_hooks.py
...
== Running post-postproc hook...
== EESSI hook to respect CUDA license not triggered for installation path /home/ocaisa/eessi/versions/2023.06/software/linux/x86_64/intel/skylake_avx512/software/CUDA/12.1.1
```